### PR TITLE
feat(UI-952): create new project - open creation modal like in templates

### PR DESCRIPTION
**Pull request by RonenMars**
## Description
When we click on Plus (new project), instead of creating a project with a random name - display a modal like we have with the templates - to get the new project name.

If a project with that name exist - display an error.

## Linear Ticket
https://linear.app/autokitteh/issue/UI-949/create-new-project-open-creation-modal-like-in-templates

## What type of PR is this? (check all applicable)

-   [x] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/components/organisms/newProjectModal.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		const { data: newProjectResponse, error } = await createProject(projectName, true);
+		setIsCreating(false);
+
+		if (error) {
+			addToast({
+				message: t("errorCreatingProject"),
+				type: "error",
+			});
+
+			return;
```
Comment (RonenMars) [inline src/components/organisms/newProjectModal.tsx:58]: add an indication of error inside the modal as well.
In this modal and in the templates too.

Comment (RonenMars) [reply src/components/organisms/newProjectModal.tsx]: done

File: src/components/organisms/newProjectModal.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import { useModalStore, useProjectStore, useToastStore } from "@src/store";
+
+import { Button, ErrorMessage, Input, Loader } from "@components/atoms";
+import { Modal } from "@components/molecules";
+
+export const NewProjectModal = () => {
+	const { t } = useTranslation("modals", { keyPrefix: "newProject" });
+	const [isCreating, setIsCreating] = useState(false);
+	const { closeModal } = useModalStore();
+	const { projectsList } = useProjectStore();
```
Comment (J1za) [inline src/components/organisms/newProjectModal.tsx:18]:
```suggestion
	const { projectsList, createProject } = useProjectStore();
```

File: src/components/organisms/newProjectModal.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	const addToast = useToastStore((state) => state.addToast);
+	const { createProject } = useProjectStore();
```
Comment (J1za) [inline src/components/organisms/newProjectModal.tsx:20]: remove

File: src/components/organisms/newProjectModal.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+	return (
+		<Modal className="w-1/2 min-w-550 p-5" hideCloseButton name={ModalName.newProject}>
+			<form onSubmit={handleSubmit(onSubmit)}>
+				<div className="mb-3 flex items-center justify-end gap-5">
+					<h3 className="mb-5 mr-auto text-xl font-bold">{t("title")}</h3>
+				</div>
+				<Input
+					label={t("projectName")}
+					placeholder="Enter project name"
```
Comment (J1za) [inline src/components/organisms/newProjectModal.tsx:78]: add to i18n

File: src/components/organisms/newProjectModal.tsx (hunk 2 above)
Comment (RonenMars) [reply src/components/organisms/newProjectModal.tsx:18]: done

File: src/components/organisms/newProjectModal.tsx (hunk 3 above)
Comment (RonenMars) [reply src/components/organisms/newProjectModal.tsx:20]: done

File: src/locales/en/modals/translation.json [hunk 5]
```diff
@@ -79,5 +80,19 @@
 		"content": "Changes might affect the currently running deployments",
 		"cancelButton": "Cancel",
 		"agreeButton": "Ok"
+	},
+	"newProject": {
+		"title": "Create new projecct",
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/locales/en/modals/translation.json]: fixed

**Review by J1za — APPROVED** [review]

**Pull request merged**